packages/tabs/src/Tabs.tsx

```diff
@@ -132,36 +133,43 @@ function Tabs({
     setSelected(index);
   }
 
-  function handleKeyDown(e: React.KeyboardEvent) {
-    e.stopPropagation();
-
-    const enabledIndexes = childrenArray.reduce((acc, child, index) => {
-      if (child.props.disabled) {
-        return acc;
-      }
-
-      return [...acc, index];
-    }, [] as Array<number>);
-
-    const enabledCurrentIndex = enabledIndexes.indexOf(selected!);
-
-    switch (e.keyCode) {
-      case keyMap.ArrowRight:
-        setSelected(
-          enabledIndexes[(enabledCurrentIndex + 1) % enabledIndexes.length],
-        );
-        break;
-
-      case keyMap.ArrowLeft:
-        setSelected(
-          enabledIndexes[
-            (enabledCurrentIndex - 1 + enabledIndexes.length) %
-              enabledIndexes.length
-          ],
-        );
-        break;
+  const enabledIndexLogic: () => [Array<number>, number] = () => {
+    const enabledIndexes: Array<number> = childrenArray.reduce(
+      (acc, child, index) => {
+        if (child.props.disabled) {
+          return acc;
+        }
+
+        return [...acc, index];
+      },
+      [] as Array<number>,
+    );
+
+    return [enabledIndexes, enabledIndexes.indexOf(selected!)];
+  };
+
+  const handleArrowKeyPress = e => {
+    if (e.keyCode === keyMap.ArrowRight) {
+      const [enabledIndexes, current] = enabledIndexLogic();
+      setSelected(enabledIndexes[(current + 1) % enabledIndexes.length]);
+    } else if (e.keyCode === keyMap.ArrowLeft) {
+      const [enabledIndexes, current] = enabledIndexLogic();
+      setSelected(
+        enabledIndexes[
+          (current - 1 + enabledIndexes.length) % enabledIndexes.length
+        ],
+      );
     }
-  }
+  };
+
+  const enabled =
+    navigator.platform.indexOf('Mac') > -1
+      ? !useKeyPress(keyMap.Command)
```

Comment:
Conditional hooks can break their behavior. Instead we should change what argument we pass into `useKeyPress` and let the hook update accordingly.

Also we should be able to pull this logic outside of the hook so we don't re-run it every time.

```
const isMac = navigator.platform.indexOf('Mac') > -1;
const keyMapValue = isMac? command : control;
```

Reply:
👍